compiler/rustc_mir_build/src/thir/cx/mod.rs

```diff
@@ -67,12 +67,6 @@ impl<'tcx> Cx<'tcx> {
 
         match self.tcx.at(sp).lit_to_const(LitToConstInput { lit, ty, neg }) {
             Ok(c) => c,
-            Err(LitToConstError::UnparseableFloat) => {
-                // FIXME(#31407) this is only necessary because float parsing is buggy
-                self.tcx.sess.span_err(sp, "could not evaluate float literal (see issue #31407)");
-                // create a dummy value and continue compiling
-                self.tcx.const_error(ty)
-            }
             Err(LitToConstError::Reported) => {
```

Comment:
I'm afraid MIR building is way outside my league, I have no familiarity with this code.
Cc @LeSeulArtichaut

Reply:
This should be good, because `PatternError::FloatBug` can't occur anymore, since it depends on `LitToConstError::UnparseableFloat`, which also can't occur. Furthermore, as described [here](https://github.com/rust-lang/rust/pull/86761#discussion_r663428484), the parser cannot fail once it has tokenized a float.